### PR TITLE
Added functionality for rendering generated graphs in the front-end

### PR DESCRIPTION
**Pull request by nielsdejong**
New features:
- In the table view, added a button 'View' that is enabled after the graph generation completes.
- The button opens up a modal, which runs a Cypher query using the selected document name.
- The result of the query is a list of nodes and relationships that are related to the document (neighbourhood of max. depth 3)
- Results are rendered as a graph visualization using the NVL library. Captions, icons and styling is inferred from the labels and properties in the graph.

### CODE REVIEW

**Review by jexp — APPROVED** [review]
Looks good to me for a per-file view

File: frontend/src/components/GraphViewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+              };
+            });
+            const newRels = neo4jRels.map(r => {
+              return { id: r.elementId, from: r.startNodeElementId, to: r.endNodeElementId, caption: r.type };
+            });
+
+            setNodes(newNodes);
+            setRelationships(newRels);
+          } else {
+            console.error("Unable to retrieve document graph for " + inspectedName, results);
```
Comment (msenechal) [inline frontend/src/components/GraphViewModal.tsx:116]: Can we have a state for error handling so we can feed that back to users with a banner saying something went wrong

File: frontend/src/components/GraphViewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    instanceId: 'inspect-graph',
+  };
+
+  return (
+    <>
+      <Dialog size='unset' open={open} aria-labelledby='form-dialog-title' disableCloseButton>
+        <Dialog.Header id='form-dialog-title'>Inspect Generated Graph from {inspectedName}. </Dialog.Header>
+        <Dialog.Content className='n-flex n-flex-col n-gap-token-4'>
+          <div style={{ width: '100%', height: '600px' }}>
+            <InteractiveNvlWrapper
```
Comment (msenechal) [inline frontend/src/components/GraphViewModal.tsx:150]: see comment above, so here we could display a danger/warning banner if there was an error while processing/getting/rendering the graph

**Review by praveshkumar1988 — APPROVED** [review]
Approved

**Review by msenechal — CHANGES_REQUESTED** [review]

**Comment by nielsdejong** [comment]
To fix the issue with overlapping icons and node captions, we can use the new `captions` property of a node (as opposed to `caption` which we currently use):
https://didactic-garbanzo-oq54v33.pages.github.io/interfaces/Core.Node.html

Then, the `captionAlign` will have an effect:
image

here's an example of how to implement this:
https://github.com/neo-technology/neo4j-nevada/blob/main/packages/examples/src/exampleModules/test-examples/ReactWrapperWithCaptionsConfExample.jsx

**Pull request merged**